index.js

```diff
@@ -622,6 +638,72 @@ var QUERIES = {
       }, [])
     }
   },
+  baseline: {
+    matches: ['year', 'availability'],
+    regexp: /^baseline\s+(?:(\d+)|(newly|widely)\s+available)$/i,
+    select: function (context, node) {
+      var since
+      var newlyAvailable = node.availability && node.availability.toLowerCase() === 'newly';
+
+      if (node.availability && node.availability.toLowerCase() === 'widely') {
+        var now = new Date();
+        now.setUTCMonth(now.getUTCMonth() - 30)
+        since = Date.UTC(now.getUTCFullYear(), now.getUTCMonth(), now.getUTCDate(), 0, 0, 0) / 1000
+      } else if (node.year) {
+        var year = parseInt(node.year)
+        since = Date.UTC(year, 11, 31, 23, 59, 0) / 1000
+      }
+
+      return Object.keys(agents).reduce(function (selected, name) {
+        if (!isBaselineCoreBrowser(name)) return selected;
+        var data = byName(name, context)
+        if (!data) return selected
+
+        var list = []
+
+        if (newlyAvailable) {
+          list = data.released.slice(-1)
+          list = list.map(nameMapper(data.name))
+          list = filterJumps(list, data.name, 1, context)
+          return selected.concat(list)
+        }
+
+        var releases = Object.keys(data.releaseDate)
+        releases.sort(function (a, b) {
+          return data.releaseDate[a] - data.releaseDate[b]
+        })
```

Comment:
Are caniuse releases always sorted?

If they are, we can omit this, but I didn't want to make this assumption.